spec/spec_helper.rb

```diff
@@ -81,7 +81,107 @@
     # (e.g. via a command-line flag).
     config.default_formatter = 'doc'
   end
+
+  # Check for leaky test resources.
+  #
+  # Execute this after the test has finished
+  # teardown and mock verifications.
+  #
+  # Changing this to `config.after(:each)` would
+  # put this code inside the test scope, interfering
+  # with the test execution.
+  config.around(:each) do |example|
+    example.run.tap do
+      # TODO: Temporarily disabled for `spec/ddtrace/workers`
+      # was meaningful changes are required to address clean
+      # teardown in those tests.
+      # They currently flood the output, making our test
+      # suite output unreadable.
+      next if example.file_path.start_with?('./spec/ddtrace/workers/async_spec.rb')
+
+      # Exclude acceptable background threads
+      background_threads = Thread.list.reject do |t|
+        group_name = t.group.instance_variable_get(:@group_name) if t.group.instance_variable_defined?(:@group_name)
+        backtrace = t.backtrace || []
+
+        # Current thread
+        t == Thread.current ||
+          # Thread has shut down, but we caught it right as it was still alive
+          !t.alive? ||
+          # Internal JRuby thread
+          defined?(JRuby) && JRuby.reference(t).native_thread.name == 'Finalizer' ||
+          # WEBrick singleton thread for handling timeouts
+          backtrace.find { |b| b.include?('/webrick/utils.rb') } ||
+          # WEBrick server thread
+          t[:WEBrickSocket] ||
+          # Rails connection reaper
+          backtrace.find { |b| b.include?('lib/active_record/connection_adapters/abstract/connection_pool.rb') } ||
+          # Ruby JetBrains debugger
+          t.class.name.include?('DebugThread') ||
+          # Categorized as a known leaky thread
+          !group_name.nil?
+      end
+
+      unless background_threads.empty?
+        info = background_threads.each_with_index.flat_map do |t, idx|
+          caller = t.instance_variable_get(:@caller) || ['(Not available. Possibly a native thread.)']
+          backtrace = t.backtrace || ['(Not available. Possibly a native thread.)']
+          [
+            "#{idx + 1}: #{t} (#{t.class.name})",
+            'Thread Creation Site:',
+            caller.map { |l| "\t#{l}" }.join("\n"),
+            'Thread Backtrace:',
+            backtrace.map { |l| "\t#{l}" }.join("\n"),
+            "\n"
+          ]
+        end.join("\n")
+
+        # We cannot fail tests gracefully in an `after(:all)` block.
+        # The test results have already been decided by RSpec.
+        # We resort to a more "blunt approach.
```

Comment:
Is this still the case? Could we break the individual tests now, using the around?

Reply:
This was stale, 🙇!